nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/ArtifactsXmlAbsoluteUrlRemover.java

```diff
@@ -77,38 +79,41 @@
 
   private static final String ARTIFACTS_XML = "artifacts.xml";
 
-  public TempBlob removeMirrorUrlFromArtifactsXml(final TempBlob artifact,
-                                                  final Repository repository,
-                                                  final String extension)
+  private static final String REPOSITORY = "repository";
+
+  public TempBlob removeMirrorUrlFromArtifactsXml(
+      final TempBlob artifact,
+      final Repository repository,
+      final String extension)
   {
     try {
       Path tempFile = createTempFile("p2-artifacts-" + randomUUID().toString(), "xml");
```

Comment:
we can use 
Path tempFile = createTempFile("p2-artifacts", "xml");
the function create a file using the formula "prefix + random_long_no + suffix"